doc/api/assert.markdown

```diff
@@ -56,47 +144,184 @@ is thrown instead.
       TypeError
     );
 
+If an `AssertionError` is thrown and a value is provided for the `message`
+parameter, the value of `message` will be appended to the `AssertionError`
+message:
+
+    assert.doesNotThrow(
+      function() {
+        throw new TypeError('Wrong value');
+      },
+      TypeError,
+      'Whoops'
+    );
+    // Throws: AssertionError: Got unwanted exception (TypeError). Whoops
+
 ## assert.equal(actual, expected[, message])
 
-Tests shallow, coercive equality with the equal comparison operator ( `==` ).
+Tests shallow, coercive equality between the `actual` and `expected` parameters
+using the equal comparison operator ( `==` ).
+
+    const assert = require('assert');
+    
+    assert.equal(1, 1);
+      // OK, 1 == 1
+    assert.equal(1, '1');
+      // OK, 1 == '1'
+      
+    assert.equal(1, 2);
+      // AssertionError: 1 == 2
+    assert.equal({a: {b: 1}}, {a: {b: 1}});
+      //AssertionError: { a: { b: 1 } } == { a: { b: 1 } }
+
+If the values are not equal, an `AssertionError` is thrown with a `message`
+property set equal to the value of the `message` parameter. If the `message`
+parameter is undefined, a default error message is assigned.
 
 ## assert.fail(actual, expected, message, operator)
 
-Throws an `AssertionError`. If `message` is falsy, it displays the values for
-`actual` and `expected` separated by the provided `operator`. Otherwise, it
-displays `message` (and does not use `actual`, `expected`, and `operator`).
+Throws an `AssertionError`. If `message` is falsy, the error message is set as
+the values of `actual` and `expected` separated by the provided `operator`.
+Otherwise, the error message is the value of `message`.
+
+    const assert = require('assert');
+    
+    assert.fail(1, 2, undefined, '>');
+      // AssertionError: 1 > 2
+    
+    assert.fail(1, 2, 'whoops', '>');
+      // AssertionError: whoops
 
 ## assert.ifError(value)
 
 Throws `value` if `value` is truthy. This is useful when testing the `error`
 argument in callbacks.
 
+    const assert = require('assert');
+    
+    assert.ifError(0); // OK
+    assert.ifError(1); // Throws 1
+    assert.ifError('error') // Throws 'error'
+    assert.ifError(new Error()); // Throws Error
+
 ## assert.notDeepEqual(actual, expected[, message])
 
 Tests for any deep inequality. Opposite of [`assert.deepEqual`][].
 
+    const assert = require('assert');
+
+    const obj1 = {
+      a : {
+        b : 1
+      }
+    };
+    const obj2 = {
+      a : {
+        b : 2
+      }
+    };
+    const obj3 = {
+      a : {
+        b : 1
+      }
+    }
+    const obj4 = Object.create(obj1);
+
+    assert.deepEqual(obj1, obj1);
+      AssertionError: { a: { b: 1 } } notDeepEqual { a: { b: 1 } }
+      
+    assert.deepEqual(obj1, obj2);
+      // OK, obj1 and obj2 are not deep equal
```

Comment:
To match adjective/adverb choice elsewhere, this should be `deeply equal` (or change any other instances of `strictly equal` and `strictly inequal` to `strict`)